src/api/forms/service.js

```diff
@@ -133,6 +134,54 @@ export async function updateDraftFormDefinition(formId, formDefinition) {
   return draftFormDefinition.create(formId, formDefinition)
 }
 
+/**
+ * Creates the live form from the current draft state
+ * @param {string} formId - ID of the form
+ * @param {FormMetadataAuthor} author - the author of the new live state
+ */
+export async function createLiveFromDraft(formId, author) {
+  // Get the form metadata from the db
+  const form = await getForm(formId)
+
+  if (!form) {
+    throw Boom.notFound(`Form with id '${formId}' not found`)
+  }
+
+  if (!form.draft) {
+    throw Boom.notFound(`Form with id '${formId}' has no draft state`)
+  }
+
+  // Build the live state
+  const now = new Date()
+  const set = !form.live
+    ? {
+        // Initialise the live state
+        live: {
+          updatedAt: now,
+          updatedBy: author,
+          createdAt: now,
+          createdBy: author
+        }
+      }
+    : {
+        // Partially update the live state
+        'live.updatedAt': now,
+        'live.updatedBy': author
+      }
+
+  // Copy the draft form definition
+  await draftFormDefinition.createLiveFromDraft(formId)
+
+  // Update the form with the live state and clear the draft
+  const result = await formMetadata.update(formId, {
+    $set: set,
+    $unset: { draft: '' }
+  })
+
+  // Return true if updated record count is 1
+  return result.modifiedCount === 1
```

Comment:
Another one here

What if two records were modified?

Do we just worry about truthy counts (non-zero) for now?

```suggestion
  // Return true if records were modified
  return !!result.modifiedCount
```

Reply:
`mongo.updateOne` is guaranteed to update only a single document.
However, if we don't update anything, that's `exception`al and indicates something going wrong with the program which we ought to know about.

Moved the throw inside the service with 7d947d8ede4b5cc5f68ee74a05a2f71dcb73415c